public/app/features/alerting/unified/components/receivers/form/ReceiverForm.tsx

```diff
@@ -0,0 +1,133 @@
+
+  const { handleSubmit, register, errors, getValues } = formAPI;
+
+  const { items, append, remove } = useControlledFieldArray<R>('items', formAPI);
+
+  const validateNameIsAvailable: Validate = useCallback(
+    (name: string) =>
+      takenReceiverNames.map((name) => name.trim().toLowerCase()).includes(name.trim().toLowerCase())
+        ? 'Another receiver with this name already exists.'
+        : true,
+    [takenReceiverNames]
+  );
+
+  return (
+    <FormContext {...formAPI}>
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <h4 className={styles.heading}>{initialValues ? 'Update contact point' : 'Create contact point'}</h4>
+        {error && (
+          <Alert severity="error" title="Error saving template">
+            {error.message || (error as any)?.data?.message || String(error)}
+          </Alert>
+        )}
+        <Field label="Name" invalid={!!errors.name} error={errors.name && errors.name.message}>
+          <Input
+            width={39}
+            name="name"
+            ref={register({ required: 'Name is required', validate: { nameIsAvailable: validateNameIsAvailable } })}
+          />
+        </Field>
+        {items.map((item, index) => {
+          const initialItem = initialValues?.items.find(({ __id }) => __id === item.__id);
+          return (
+            <ChannelSubForm<R>
+              key={item.__id}
+              onDuplicate={() => {
+                const currentValues = getValues({ nest: true }).items[index];
+                append({ ...currentValues, __id: String(Math.random()) });
+              }}
+              onDelete={() => remove(index)}
+              pathPrefix={`items.${index}.`}
+              notifiers={notifiers}
+              secureFields={initialItem?.secureFields}
+              errors={errors?.items?.[index] as NestDataObject<R, FieldError>}
+              commonSettingsComponent={commonSettingsComponent}
+            />
+          );
+        })}
+        <Button type="button" icon="plus" onClick={() => append({ ...defaultItem, __id: String(Math.random()) } as R)}>
+          New contact point type
+        </Button>
+        <div className={styles.buttons}>
+          {loading && (
+            <Button disabled={true} icon="fa fa-spinner" variant="primary">
+              Saving...
+            </Button>
+          )}
+          {!loading && <Button type="submit">Save contact point</Button>}
+          <LinkButton
+            disabled={loading}
+            variant="secondary"
+            href={makeAMLink('/alerting/notifications', alertManagerSourceName)}
+          >
+            Cancel
+          </LinkButton>
+        </div>
+      </form>
+    </FormContext>
+  );
+}
+
+const getStyles = (theme: GrafanaThemeV2) => ({
+  heading: css`
+    margin: ${theme.spacing(4, 0)};
+  `,
+  buttons: css`
+    margin-top: ${theme.spacing(4)};
+
+    & > * + * {
+      margin-left: ${theme.spacing(1)};
+    }
```

Comment:
what is this for?

Reply:
adds spacing between child nodes (buttons), so they're not glued together

Reply:
Can you use a HorizonalGroup instead?

Reply:
hah, did not know this exists. Thanks, will try
- [ ] use HorizontalGroup